packages/common/container-definitions/src/runtime.ts

```diff
@@ -131,11 +131,15 @@ export interface IBatchMessage {
  * loader layer.
  * It gets passed into the {@link (IRuntimeFactory:interface).instantiateRuntime} call.
  *
- * @privateremarks
+ * @privateRemarks
  * Only include members on this interface if you intend them to be consumed/called from the runtime layer.
  *
- * TODO: once `@alpha` tag is removed, `unknown` should be removed from submitSignalFn.
- * See {@link https://dev.azure.com/fluidframework/internal/_workitems/edit/7462}
+ * TODO: once `@internal` or `@system`, `submitSignalFn`'s `contents` `unknown` type should be replaced with
+ * {@link @fluidframework/core-interfaces#ISignalEnvelope}.
+ * See {@link https://dev.azure.com/fluidframework/internal/_workitems/edit/7462}.
+ *
+ * To get `IContainerContext` internal, likely externally exposed uses will need
```

Comment:
I don't think we're on a track to make `IContainerContext` internal, but rather I think an equivalent eventually may become part of the public API.  This is (currently) required for the extensibility point of a customer-defined `IRuntimeFactory`, which is actually a useful tool.  The interfaces could do with a lot of improvements, but the core concept seems valuable.